packages/saltcorn-data/tests/table.test.ts

```diff
@@ -2596,3 +2597,279 @@ describe("grandparent join", () => {
 });
+
+// Testing slug_options method
+describe("Table slug options", () => {
+  it("should return slug options for unique string fields", async () => {
+    const table = await Table.create("slug_test_table");
+    await Field.create({
+      table,
+      name: "title",
+      label: "Title",
+      type: "String",
+      is_unique: true,
+    });
+    await Field.create({
+      table,
+      name: "description",
+      label: "Description",
+      type: "String",
+    });
+
+    const options = await table.slug_options();
+    expect(options).toEqual([
+      { label: "", steps: [] },
+      {
+        label: "/:id",
+        steps: [{ field: "id", unique: true, transform: null }],
+      },
+      {
+        label: "/slugify-title",
+        steps: [{ field: "title", unique: true, transform: "slugify" }],
+      },
+    ]);
+  });
+
+  it("should return the default option with id if no unique string fields exist", async () => {
+    const table = await Table.create("slug_test_table_no_unique");
+    await Field.create({
+      table,
+      name: "description",
+      label: "Description",
+      type: "String",
+    });
+    await Field.create({
+      table,
+      name: "age",
+      label: "Age",
+      type: "Integer",
+      is_unique: false,
+    });
+
+    const options = await table.slug_options();
+    expect(options).toEqual([
+      { label: "", steps: [] },
+      {
+        label: "/:id",
+        steps: [{ field: "id", unique: true, transform: null }],
+      },
+    ]);
+  });
+
+  it("should handle tables with no fields", async () => {
+    const table = await Table.create("slug_test_table_empty");
+    const options = await table.slug_options();
+    expect(options).toEqual([
+      { label: "", steps: [] },
+      {
+        label: "/:id",
+        steps: [{ field: "id", unique: true, transform: null }],
+      },
+    ]);
+  });
+
+  it("should handle tables with non-string unique fields", async () => {
+    const table = await Table.create("slug_test_table_non_string");
+    await Field.create({
+      table,
+      name: "age",
+      type: "Integer",
+      is_unique: true,
+    });
+    await Field.create({
+      table,
+      name: "created_at",
+      type: "Date",
+      is_unique: true,
+    });
+
+    const options = await table.slug_options();
+    expect(options).toEqual([
+      { label: "", steps: [] },
+      {
+        label: "/:id",
+        steps: [{ field: "id", unique: true, transform: null }],
+      },
+      {
+        label: "/:age",
+        steps: [{ field: "age", unique: true, transform: null }],
+      },
+      {
+        label: "/:created_at",
+        steps: [{ field: "created_at", unique: true, transform: null }],
+      },
+    ]);
+  });
+});
+
+describe("ownership options", () => {
+  it("should return user reference fields", async () => {
+    await Table.create("test_table");
+    const table = Table.findOne({ name: "test_table" });
+    assertIsSet(table);
+
+    table.fields = [
```

Comment:
Use Field.create like the other tests. It is slower but less code;

Reply:
I removed it here since there is a similar test case in auth.test.ts